src/util/native_fs_utils.js

```diff
@@ -199,10 +199,17 @@ async function safe_move_posix(fs_context, src_path, dst_path, src_ver_info, tmp
 // safe_link_posix links src_path to dst_path while verifing dst_path has the expected ino and mtimeNsBigint values
 // src_file exists on uploads (open mode = 'w' ) or deletions
 // on uploads (open mode 'wt') the dir_file is used as the link source
-async function safe_move_gpfs(fs_context, src_path, dst_path, gpfs_options) {
-    const { src_file = undefined, dst_file = undefined, dir_file = undefined, should_unlink = false } = gpfs_options;
+async function safe_move_gpfs(fs_context, src_path, dst_path, gpfs_options, src_ver_info) {
+    const { src_file = undefined, dst_file = undefined, dir_file = undefined, should_unlink = false,
+        is_move_to_versions = false } = gpfs_options;
     dbg.log1('Namespace_fs.safe_move_gpfs', src_path, dst_path, dst_file, should_unlink);
-    await safe_link_gpfs(fs_context, dst_path, src_file || dir_file, dst_file);
+    if (is_move_to_versions) {
```

Comment:
I'm not convinced this function should be aware of the destination... 
also you used safe_move_posix() directly, so why doing this change?

Reply:
on delete we call safe_move_posix() directly because we also want to unlink the source. but on move_to_dest we dont want to unlink because the file we be overwritten by the move to latest. 
The reason I have this if here instead of in _move_to_dest_version is because its seemed to be in line with how we use safe_move in combination with open_files_gpfs. where open_files_gpfs decides how the move should be handled (e.g should we verify the dst file, should we unlink ), and safe_move_gpfs does the move